src/nanoarrow/common/array.c

```diff
@@ -115,7 +115,7 @@ static ArrowErrorCode ArrowArraySetStorageType(struct ArrowArray* array,
       break;
     case NANOARROW_TYPE_BINARY_VIEW:
     case NANOARROW_TYPE_STRING_VIEW:
-      array->n_buffers = NANOARROW_BINARY_VIEW_FIXED_BUFFERS;
+      array->n_buffers = NANOARROW_BINARY_VIEW_FIXED_BUFFERS + 1;
```

Comment:
While this "works" I'm not sure if we still care to maintain this distinction of `NANOARROW_BINARY_FIXED_VIEW_BUFFERS` being 2 when all other types really have 3 fixed buffers. I suppose the last buffer isn't fixed, in the sense that it can be moved. So semantically maybe there is some value in having this still? But we could probably reconsolidate some code if we go back to just using `NANOARROW_MAX_FIXED_BUFFERS` like the other types